spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/Engine.scala

```diff
@@ -88,15 +90,12 @@ object Engine {
    */
   def init: Unit = this.synchronized {
     if (localMode) {
-      require(getSparkMaster == null, "Detect BIGDL_LOCAL_MODE and spark.master are both set." +
-        " They're conflict. Please reset BIGDL_LOCAL_MODE if you're use Spark or remove " +
-        "spark.master if you're run without spark")
       logger.info("Detect BIGDL_LOCAL_MODE is set. Run workload without spark")
       // The physical core number should have been initialized by env variable in bigdl.sh
-      setNodeAndCore(1, getCoreNumberFromEnv)
+      setNodeAndCore(1, getNumMachineCores)
```

Comment:
@yiheng I'm still not so sure how the local mode differs from Spark local mode, since this uses Spark in all cases and Spark local is in one JVM, but this is a separate topic, maybe. See https://github.com/intel-analytics/BigDL/pull/827#issuecomment-298637426 for example. It seems like you can still set a Spark master in this case to control the configure Engine cores, even if some code will not actually use Spark.

Is the ask here to put back the `DL_CORE_NUMBER` property? that's fine by me, whatever would make this easier to digest. More changes could happen later.